src/librustdoc/visit_ast.rs

```diff
@@ -281,31 +285,33 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
         };
 
         let is_private = !self.cx.cache.effective_visibilities.is_directly_public(tcx, ori_res_did);
-        let is_hidden = inherits_doc_hidden(tcx, res_did, None);
-
-        // Only inline if requested or if the item would otherwise be stripped.
-        if (!please_inline && !is_private && !is_hidden) || is_no_inline {
-            return false;
-        }
-
-        if !please_inline &&
-            let Some(item_def_id) = reexport_chain(tcx, def_id, res_did).iter()
+        let is_hidden = tcx.is_doc_hidden(ori_res_did);
+        let item = tcx.hir().get_by_def_id(res_did);
+        let is_module = matches!(item, Node::Item(&hir::Item { kind: hir::ItemKind::Mod(_), .. }));
```

Comment:
Because if it's a module, it can contain items that could be re-exported and we need to to go through them too. A good example of this can be seen in `tests/rustdoc/infinite-redirection.rs`.

Reply:
I added a comment explaining it and moved this `matches!()` directly into the `if` condition.